Week02/Assignment/afilbrun_01.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 18 14:16:58 2018
+
+@author: amandl
+"""
+
+# import python libraries and provided data 
+
+import numpy as np
+import scipy as sp
+import matplotlib.pyplot as plt
+
+x = np.array([ 1.,   1.5,  2.,   2.5,  3.,   3.5,  4.,   4.5,  5.,   5.5,  6.,   6.5,  7.,   7.5, 8.,   8.5,  9.,   9.5, 10. ])
+y = np.array([3.43, 4.94, 6.45, 9.22, 6.32, 6.11, 4.63, 8.95, 7.8, 8.35, 11.45, 14.71, 11.97, 12.46, 17.42, 17.0, 15.45, 19.15, 20.86])
+
+# 1 & 2 - Linear and Cubic Curve Fitting
+
+from scipy.stats import linregress
+slope, intercept, r_value, p_value, std_err = linregress(x,y)
+
+coefficients = np.polyfit(x, y, 3)
+pfit = np.poly1d(coefficients)
```

Comment:
This is a legitimate way to fit a polynomial function, but could you go back and fit the function using scipy rather than numpy? Have a look at the curve_fit function we discussed.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html